### DISCUSSION TITLE
Adapt placeholder formats based on the SQL dialect

### DISCUSSION THREAD

**Discussion by PrettyWood** [PRQL/prql#4969] [Ideas]
Hello everyone.

Sorry if I didn't find an issue already opened regarding this topic.
Currently
```prql
from employees
filter id == $1
```
will always be compiled into
```sql
SELECT * FROM employees WHERE id = $1
```
even if the dialect is gbq or sqlite.
It would be great if it could compile to 
```sql
SELECT * FROM employees WHERE id = ?
```
for example.

Of course there is the issue of positional vs named params. Postgres doesn't support named params so maybe we should support only positional params for now or support also `$name` but panic if translated to postgres for example
WDYT?

**Comment by max-sixty** [comment]
Yes, good point. 

At the moment we don't really have a "param" format in PRQL — we just pass everything through.
Definitely open to establishing one! Or at least triaging by translating `$n` to something that works in all dialects.

With that proposal, how would we handle multiple `$n` — presumably because postgres is positional, multiple `$1` would be translated to multiple `?`, which postgres would expect multiple arguments for.

Could we use Postgres's named params? I'm looking at https://stackoverflow.com/questions/7389416/postgresql-how-to-pass-parameters-from-command-line, seems to suggest `:'v1'` might work?
